diff -r 4a00917c4d5f linux-2.6.26/include/asm-x86/hardirq_64.h
--- a/linux-2.6.26/include/asm-x86/hardirq_64.h	Thu Aug 07 23:57:46 2008 +0200
+++ b/linux-2.6.26/include/asm-x86/hardirq_64.h	Sat Aug 09 17:49:46 2008 -0700
@@ -1,5 +1,7 @@
 #ifndef __ASM_HARDIRQ_H
 #define __ASM_HARDIRQ_H
+/* For CONFIG_NSC, just ignore this stuff... Fall back... */
+#ifndef CONFIG_NSC
 
 #include <linux/threads.h>
 #include <linux/irq.h>
@@ -19,5 +21,25 @@
 #define or_softirq_pending(x)  or_pda(__softirq_pending, (x))
 
 extern void ack_bad_irq(unsigned int irq);
+#else // NSC
 
+#include <linux/threads.h>
+#include <linux/irq.h>
+
+typedef struct {
+        unsigned int __softirq_pending;
+        unsigned long idle_timestamp;
+        unsigned int __nmi_count;       /* arch dependent */
+        unsigned int apic_timer_irqs;   /* arch dependent */
+} ____cacheline_aligned irq_cpustat_t;
+
+DECLARE_PER_CPU(irq_cpustat_t, irq_stat);
+extern irq_cpustat_t irq_stat[];
+
+#define __ARCH_IRQ_STAT
+#define __IRQ_STAT(cpu, member) (per_cpu(irq_stat, cpu).member)
+
+#include <linux/irq_cpustat.h>
+
+#endif // CONFIG_NSC
 #endif /* __ASM_HARDIRQ_H */
